src/librustc_ast_lowering/item.rs

```diff
@@ -76,6 +77,24 @@ impl<'a> Visitor<'a> for ItemLowerer<'a, '_, '_> {
         }
     }
 
+    fn visit_fn(&mut self, fk: FnKind<'a>, _: Span, _: NodeId) {
+        match fk {
+            FnKind::Fn(FnCtxt::Foreign, _, sig, _, _) => {
+                self.visit_fn_header(&sig.header);
+                visit::walk_fn_decl(self, &sig.decl);
+            }
+            FnKind::Fn(_, _, sig, _, body) => {
+                self.visit_fn_header(&sig.header);
+                visit::walk_fn_decl(self, &sig.decl);
+                walk_list!(self, visit_block, body);
+            }
+            FnKind::Closure(decl, body) => {
+                visit::walk_fn_decl(self, decl);
+                self.visit_expr(body);
+            }
```

Comment:
Can you use `walk_fn` for this part? So that it's clearer what's being overriden?

Reply:
yeah for sure